hal/network/api/ifapi.h

```diff
@@ -247,9 +247,25 @@ typedef struct if_event_power_state if_req_power;
 typedef enum if_req_t {
     IF_REQ_NONE        = 0,
     IF_REQ_POWER_STATE = 1,
-    IF_REQ_DHCP_STATE  = 2
+    IF_REQ_DHCP_STATE  = 2,
+    IF_REQ_DRIVER_SPECIFIC = 3,
 } if_req_t;
 
+typedef struct if_req_driver_specific {
+    uint32_t type;
+} if_req_driver_specific;
```

Comment:
Why do we need this struct? Cant we just add a `type` field to `if_wiznet_pin_remap` below?

Reply:
It's so you can dereference the pointer and check the `type`, before you dereference the pointer as the unique struct.  I didn't end up using it the way it was originally intended, but if we ever add more unique types it will be handy to convert to using the `if_req_driver_specific` struct to check the type first, since `type` will always be the first 4 bytes.